src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,186 @@
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        user = new User();
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+
+        assertEquals("User is null.", exception.getMessage());
+    }
+
+    @Test
+    void registerLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Login can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Password can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerUserAgeNull_NotOk() {
+        user.setLogin("validLog");
+        user.setPassword("validPwd");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordExists_NotOk() throws RegistrationException {
+        User user = new User();
+        user.setLogin("existingUser");
+        user.setPassword("password123");
+        user.setAge(21);
+        storageDao.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() throws RegistrationException {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertDoesNotThrow(() -> registrationService.register(user));
```

Comment:
According to the checklist, it is important to check all possible edge cases. In this test case, you're testing invalid login lengths but not the edge cases for valid login lengths. Make sure to include a test for a login with exactly 6 characters, which is the minimum valid length, to ensure the edge case is handled correctly.